biojava-structure/src/main/java/org/biojava/nbio/structure/ChainImpl.java

```diff
@@ -592,8 +592,10 @@ public String getSeqResSequence(){
 					PolymerType.POLYNUCLEOTIDE_ONLY.contains(cc.getPolymerType())){
 				// an amino acid residue.. use for alignment
 				String oneLetter= ChemCompGroupFactory.getOneLetterCode(cc);
-				if ( oneLetter == null || oneLetter.isEmpty() || oneLetter.equals("?"))
+				// AB oneLetter.length() should be one. e.g. in 1EMA it is 3 and this makes mapping residue to sequence impossible.
+				if ( oneLetter == null || oneLetter.isEmpty() || oneLetter.equals("?") || oneLetter.length()!=1) {
```

Comment:
FYI this changes the behaviour discussed in #457, with this `getSeqResSequence` and `getSeqResGroups` have the same size